lib/crossdoc/tree.rb

```diff
@@ -36,11 +36,31 @@ def download
         elsif @src.index('./')==0
           @io = open(@src.gsub('./', Dir.pwd + '/'))
         else # assume it's a URL
-          @io = URI.open(@src)
+          @io = URI.open @src
+          if @is_svg || num_images < 6
+            return
+          end
+          img = MiniMagick::Image.open @src
+          img.combine_options do |i|
+            i.quality 80
+            i.geometry image_width
+            i.background '#FFFFFF'
+            i.alpha 'remove'
+          end
+          img.format 'jpg'
+          @io = URI.open img.path
         end
       end
     end
 
+    # decrease width as image file size increases to decrease final pdf size
+    # starts from 2 x page width (1024px)
+    # 1MB -> 963px, 6MB -> 658px
+    def image_width
```

Comment:
If we can pass the size in as a parameter I would prefer that. I think you can use `i.size` on the object that's passed into the combine_options block

Reply:
That doesn't seem to be an option, since i is some sort of object with limited features from img. I think it only has setters, not getters.

![CleanShot 2024-04-12 at 13 13 46](https://github.com/TinyMission/crossdoc/assets/25541844/a7947564-19cb-4dae-9dc0-7a58b16dc738)

![CleanShot 2024-04-12 at 13 13 30](https://github.com/TinyMission/crossdoc/assets/25541844/f03420d2-3101-40ee-a07d-db73a319ff98)

I can still make it a param like this, though.
```img = MiniMagick::Image.open @src
img_size = img.size
img.combine_options do |i|
  i.geometry image_width(img_size)
  ...
end

Reply:
Oh, yeah that works 👍